akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphMatValueSpec.scala

```diff
@@ -235,6 +235,25 @@ class GraphMatValueSpec extends StreamSpec {
       m4 should ===(NotUsed)
     }
 
+    "build more complicated graph with flows optimized for identity flows" in {
```

Comment:
Following the test case right above, but use `fromSinkAndSourceMat` as the #22899 issue is realized when building more complicated graphs.